pkg/internal/kube/store.go

```diff
@@ -388,12 +390,20 @@ func (s *Store) Subscribe(observer meta.Observer) {
 	defer s.access.RUnlock()
 	s.BaseNotifier.Subscribe(observer)
 	for _, pod := range s.podsByContainer {
-		observer.On(&informer.Event{Type: informer.EventType_CREATED, Resource: pod})
+		if err := observer.On(&informer.Event{Type: informer.EventType_CREATED, Resource: pod}); err != nil {
+			s.log.Debug("observer failed. Unsubscribing it", "observer", observer.ID(), "error", err)
+			s.BaseNotifier.Unsubscribe(observer)
+			return
+		}
 	}
 	// the IPInfos could contain IPInfo data from Pods already sent in the previous loop
 	// is the subscriber the one that should decide whether to ignore such duplicates or
 	// incomplete info
 	for _, ips := range s.objectMetaByIP {
-		observer.On(&informer.Event{Type: informer.EventType_CREATED, Resource: ips})
+		if err := observer.On(&informer.Event{Type: informer.EventType_CREATED, Resource: ips}); err != nil {
+			s.log.Debug("observer failed. Unsubscribing it", "observer", observer.ID(), "error", err)
```

Comment:
is it worth to say in this line that is the observer for IPs? basically to differentiate from the previous logger